src/common.speech/ConversationTranslator/ConversationTranslatorRecognizer.ts

```diff
@@ -104,27 +91,118 @@ export class ConversationTranslatorRecognizer extends Recognizer implements ICon
         return this.privProperties;
     }
 
+    public isDisposed(): boolean {
+        return this.privIsDisposed;
+    }
+
+    /**
+     * Connect to the recognizer
+     * @param token
+     */
+    public connect(token: string, cb?: () => void, err?: (e: string) => void): void {
+        try {
+            Contracts.throwIfDisposed(this.privIsDisposed);
+            Contracts.throwIfNullOrWhitespace(token, "token");
+            this.privReco.conversationTranslatorToken = token;
+            this.privReco.connect();
+            if (!!cb) {
+                try {
+                    cb();
+                } catch (e) {
+                    if (!!err) {
+                        err(e);
+                    }
+                }
+            }
+        } catch (error) {
+            if (!!err) {
+                if (error instanceof Error) {
+                    const typedError: Error = error as Error;
+                    err(typedError.name + ": " + typedError.message);
+                } else {
+                    err(error);
+                }
+            }
+        }
+    }
+
+    /**
+     * Disconnect from the recognizer
+     */
+    public disconnect(cb?: () => void, err?: (e: string) => void): void {
+        try {
+            Contracts.throwIfDisposed(this.privIsDisposed);
+            this.privRoom = undefined;
+            this.privReco.disconnect();
```

Comment:
Ditto about synchronous

Reply:
This was based on the DialogService implementation.

Reply:
Updated to use Ralph's Async method addition.